src/test-apps/MockDMPublisher.h

```diff
@@ -46,7 +48,9 @@ class MockDMPublisher : public nl::Weave::Profiles::DataManagement::DMPublisher
 
     WEAVE_ERROR UpdateIndication(ExchangeContext *aResponseCtx, ReferencedTLVData &aDataList);
 
-    void IncompleteIndication(const uint64_t &aPeerNodeId, StatusReport &aReport);
+    virtual void IncompleteIndication(const uint64_t &aPeerNodeId, StatusReport &aReport);
+
+    using DMPublisher::IncompleteIndication;
```

Comment:
I don't quite understand.  I believe the the intention here was to have the method above override the `DMPublisher::IncompleteIndication`; Arguably, it would be appropriate to add the `__OVERRIDE` macro to the `IncompleteIndication`.  Because the `IncompleteIndication` already is declared virtual in base, there is no need to declare it virtual here, right?

Reply:
I agree, this seems an odd warning. As Jay and I discussed this morning, it might make sense long-term to get rid of "-Wall" in configure.ac and select those warnings we feel actually make sense and add real, measurable value.

Using __OVERRIDE here might be better though that may then trigger -Winconsistent-missing-override. I'll give it a try.

Reply:
The using statement here seems to be the required magic but it doesn't make much sense to me based on my knowledge of the language.

The better approach here might be just to disable -Woverloaded-virtual (-Wno-overloaded-virtual) for clang. Thoughts @robszewczyk or @jaylogue?

Reply:
IncompleteIndication() has a different signature on the base class:
`void IncompleteIndication(Binding *aBinding, StatusReport &aReport)`
vs.
`void IncompleteIndication(const uint64_t &aPeerNodeId, StatusReport &aReport)`
This is likely erroneous, as nothing appears to call the subclass method.

Reply:
Looking at this again, `src/lib/profiles/data-management/Legacy/ProtocolEngine.h` defines both:

```
virtual void IncompleteIndication(Binding *aBinding, StatusReport &aReport);
```

and:

```
virtual void IncompleteIndication(const uint64_t &aPeerNodeId, StatusReport &aReport) = 0;
```

The implementation of the former calls the latter making the base class abstract. It seems like the test clients correctly override the latter which seems to imply that the warning is a bit pedantic.